upstream/upstream.go

```diff
@@ -625,7 +625,19 @@ func (u *Upstream) detectFeatures() error {
 			u.supportedNetworkIds[util.EvmNetworkId(cfg.Evm.ChainId)] = true
 			u.supportedNetworkIdsMu.Unlock()
 		}
-		// TODO evm: check full vs archive node
+
+		if cfg.Evm.NodeType == "" {
+			nodeType, err := u.detectNodeType(&common.NormalizedRequest{})
```

Comment:
Any reason to create the req here? It can be done in isolation within "detectNodeType"